### PR TITLE
Add `useDataGrid` Hook Documentation and Example

### PR DESCRIPTION
**Pull request by salihozdemir**
Test me! 'MUI'
[Link to USE-DATA-GRID-DOC](https://use-dat-refine.pankod.com)

Hey, I'm client! Test me! 'MUI'
[Link to USE-DATA-GRID-DOC](https://use-dat-client.refine.pankod.com)

Test me! 'MUI'
[Link to USE-DATA-GRID-DOC](https://use-dat-refine.pankod.com)

Hey, I'm client! Test me! 'MUI'
[Link to USE-DATA-GRID-DOC](https://use-dat-client.refine.pankod.com)

Test me! 'MUI'
[Link to USE-DATA-GRID-DOC](https://use-dat-refine.pankod.com)

Add `useDataGrid` hook documentation for `@pankod/refine-mui` package.

Test documentation! 'MUI'
[Link to USE-DATA-GRID-DOC](https://use-dat-doc-refine.pankod.com)

### CODE REVIEW

File: documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+};
+```
+
+Above, you can see the filtering properties from `dataGridProps`.
+
+:::note
+To see how the filtering works, you can look at the [source code][source-code] of the `useDataGrid` hook.
+:::
+
+:::tip
```
Comment (ozkalai) [inline documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md:295]: Duplicated tip

**Review by biskuvit — APPROVED** [review]

File: documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+        "content": "Voluptas consequatur quia beatae. Ipsa est qui culpa deleniti...",
+        "status": "draft"
+    }
+]
+```
+
+To see basic usage, let's create a table with the `id`, `title` and `content` columns.
+
+```tsx title="/src/pages/posts/list.tsx"
+import { useDataGrid, DataGrid, GridColumns, List } from "@pankod/refine-mui";
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md:34]:
```suggestion
import React from "react";
import { useDataGrid, DataGrid, GridColumns, List } from "@pankod/refine-mui";
```

File: documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+                <Button onClick={() => handleSorting("asc")}>Asc</Button>
+                <Button onClick={() => handleSorting("desc")}>Desc</Button>
+            </ButtonGroup>
+            <DataGrid {...dataGridProps} autoHeight />
+        </List>
+    );
+};
+```
+
+Mui X community version only sort the rows according to one criterion at a time. To use multi-sorting, you need to upgrade to the [Pro plan](#).
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md:242]:
```suggestion
Mui X community version only sorts the rows according to one criterion at a time. To use multi-sorting, you need to upgrade to the [Pro plan](https://mui.com/pricing/).
```

File: documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+
+| Key                                                          | Description                                                                                                                                                        | Type                                                                           | Default                                                                              |
+| ------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------ |
+| columns<div className=" required">Required</div>             | Set of columns of type `GridColumns`                                                                                                                               | [`GridColDef`](https://mui.com/x/api/data-grid/grid-col-def)                   |                                                                                      |
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md:390]:
```suggestion
|  <div className="required-block"><div>columns</div> <div className=" required">Required</div></div>             | Set of columns of type `GridColumns`                                                                                                                               | [`GridColDef`](https://mui.com/x/api/data-grid/grid-col-def)                   |                                                                                      |
```

File: documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+| resource                                                     | The resource to use for table data                                                                                                                                 | `string` \| `undefined`                                                        | Resource name that it reads from the url                                             |
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md:391]:
```suggestion
| resource                                                     | The resource to use for table data                                                                                                                                 | `string` \| `undefined`                                                        | Resource name that it reads from the URL                                             |
```

File: documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md [hunk 6]
```diff
@@ -0,0 +1,465 @@
+---
+id: useDataGrid
+title: useDataGrid
+---
+
+By using `useDataGrid`, you are able to get properties that are compatible with MUI X `<DataGrid>` component. All features such as sorting, filtering and pagination comes as out of box. For all the other features, you can refer to the MUI X [`<DataGrid>`][data-grid] documentation.
+
+This hook is extended from [`useTable`](/core/hooks/useTable.md) from the [`@pankod/refine-core`](https://github.com/pankod/refine/tree/master/packages/core) package.
+
```
Comment (omeraplak) [inline documentation/docs/ui-frameworks/mui/hooks/useDataGrid.md:9]: Shall we add a `tip box` like "We support the Pro version" here?

File: examples/table/mui/advancedTable/public/index.html [hunk 7]
```diff
@@ -23,7 +23,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-        <title>refine mui table example</title>
+        <title>refine mui da example</title>
```
Comment (ozkalai) [inline examples/table/mui/advancedTable/public/index.html:26]: Is that a contraction?

**Review by biskuvit — APPROVED** [review]

**Comment by aliemir** [comment]
Closing this PR and reopening in #1877 for a clean commit tree

**Pull request closed without merging**